pkg/daemon/agent/agent.go

```diff
@@ -62,7 +63,27 @@ func (a *Agent) Run() error {
 		volumeManager,
 	)
 
-	flexvolumeServer.Start()
+	err = rpc.Register(flexvolumeController)
+	if err != nil {
+		return fmt.Errorf("unable to register rpc: %v", err)
+	}
+
+	driverName, err := flexvolume.RookDriverName(a.context)
+	if err != nil {
+		return fmt.Errorf("failed to get driver name. %+v", err)
+	}
+
+	err = flexvolumeServer.Start(driverName)
+	if err != nil {
+		return fmt.Errorf("failed to start flex volume server %s, %+v", driverName, err)
+	}
+
+	// Register drivers both with the name of the namespace and the legacy name "rook" in case
+	// there are still volume plugins based on the legacy plugin.
+	err = flexvolumeServer.Start(flexvolume.LegacyFlexDriverName)
```

Comment:
if the `flexvolume.RookDriverName` function had returned the legacy driver name, wouldn't we be starting 2 flexvolume servers using the same name?

Reply:
rather than check for an equivalent name, `Start()` will simply return success if the server is already running.